### ISSUE TITLE
line-pattern image error

### ISSUE THREAD

**Issue by Yuki-Zhao** [maplibre/maplibre-gl-js#2237]
add image to map，
then，set image as line-pattern。
Line render symbol with an extra line on it

demo：https://codesandbox.io/s/eloquent-einstein-o545q4?file=/index.html。
![image](https://user-images.githubusercontent.com/22816207/222069008-d8fb7a8e-9e89-4fba-98f9-c00630889da3.png)

**Comment by HarelM** [comment]
See here:
https://uploads.codesandbox.io/uploads/user/cff2d3ff-a033-4f37-a9d2-76b7f70a3add/RVSz-964460.png
This is how this image looks in the browser standalone tab:
![image](https://user-images.githubusercontent.com/3269297/222111750-e352fe45-c8cd-46f3-9718-e9f57254391f.png)
I guess is it related to how the image is configured, background etc.
I agree the outcome is not ideal, but I think there are some issue with the image as well.
I would focus on fixing those first.
I can't say I can explain the lines around the image, but I would try and play with the image background, color, transparency etc, and see what are the results.

**Comment by Yuki-Zhao** [comment]
What other information do you need?

**Comment by HarelM** [comment]
The result of the above request to try and manipulate the image and see what happens and if it solves your issue.

**Comment by Yuki-Zhao** [comment]
If I had used 964460-larger.png, it would have been fine。
But there's another problem， the line display symbol is the mirror display。

ps: the height of 964460-larger.png is 50；

**Comment by HarelM** [comment]
There's a note in the style-spec about the fact that images are best shown when they are with the size of 2,4,8.
I would also try and see if a symbol layer would work better here instead of a pattern.
>the line display symbol is the upside down

This is related to the direction of the line I believe.

**Comment by HarelM** [comment]
Yes, this is expected.
Consider a cliff where you want to show the user which part is the way down.
It should rotate with the mountain and is a common practice when talking about contours.

**Comment by qiulinFV** [comment]
> The result of the above request to try and manipulate the image and see what happens and if it solves your issue.


using this image for pattern:
![image](https://user-images.githubusercontent.com/48538029/232476040-3daeeb73-7d9a-458e-8ee4-7c84abc374a6.png)


Expected Behavior
![image](https://user-images.githubusercontent.com/48538029/232476496-e31d683c-f4f3-4259-9c47-d61dbb087e96.png)


Actual Behavior
![image](https://user-images.githubusercontent.com/48538029/232476816-11ff0076-c583-4c4f-9783-ca95a3cc22d8.png)


example :   https://codesandbox.io/s/empty-tdd-d0i0q8?file=/index.html

Am I doing something wrong? or is it a known problem?

**Comment by qiulinFV** [comment]
I noticed this code: 
https://github.com/mapbox/mapbox-gl-js/blob/main/src/render/image_atlas.js
![image](https://user-images.githubusercontent.com/48538029/232484916-41a970c5-fd48-499b-931d-d5fb934ced45.png)



I really want to know why this is necessary .

**Comment by HarelM** [comment]
You'll need to investigate into the git history, or change it and see which tests fail...

**Comment by qiulinFV** [comment]
> You'll need to investigate into the git history, or change it and see which tests fail...

Thank you very much for your reply. I'll try。
Besides，
example : https://codesandbox.io/s/empty-tdd-d0i0q8?file=/index.html
Am I doing something wrong? or is it a known problem?
Do you have any suggestions regarding the issues in the above example?

**Comment by HarelM** [comment]
I don't think you are doing something wrong, It's probably a bug in the code... Image size should be a 2,4,8, etc, but your image is 16x16, so it should work fine I believe...
